src/python/aqueduct_executor/operators/connectors/tabular/s3.py

```diff
@@ -1,14 +1,55 @@
 import io
 import json
-from typing import List
+import os
+import uuid
+from typing import List, Optional
 
 import boto3
 import pandas as pd
-from aqueduct_executor.operators.connectors.tabular import common, config, connector, extract, load
+from aqueduct_executor.operators.connectors.tabular import common, connector, extract, load
+from aqueduct_executor.operators.connectors.tabular.config import S3Config, S3CredentialType
 
 
 class S3Connector(connector.TabularConnector):
-    def __init__(self, config: config.S3Config):
+    def _handle_config_file_path(self, config: S3Config) -> None:
+        """
+        _handle_config_file_path updates config's access_key and secret_access_key
+        based on credentials in config_file_path and config_file_profile.
+        """
+        os.environ["AWS_SHARED_CREDENTIALS_FILE"] = config.config_file_path
+        os.environ["AWS_CONFIG_FILE"] = config.config_file_path
+        session = boto3.Session(profile_name=config.config_file_profile)
+        credentials = session.get_credentials()
+        config.access_key_id = credentials.access_key
+        config.secret_access_key = credentials.secret_key
+
+    def _handle_config_file_content(self, config: S3Config) -> None:
+        """
+        _handle_config_file_content updates config's access_key and secret_access_key
+        based on credentials in config_file_path and config_file_profile.
+        """
+        # write to temp file assuming the cwd is safe to create such file.
+        temp_path = os.path.join(os.getcwd(), str(uuid.uuid4()))
+        with open(temp_path, "w") as w:
+            w.write(config.config_file_content)
+        config.config_file_path = temp_path
+
+        try:
+            self._handle_config_file_path(config)
+            os.remove(temp_path)
```

Comment:
nit: you can move this cleanup task to a `finally` block: https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions